pkg/k8s/client.go

```diff
@@ -468,3 +489,99 @@ func (c *Client) GetMeshConfig() configv1alpha2.MeshConfig {
+	var policies []*policyv1alpha1.Egress
+
+	for _, egressIface := range c.informers.List(osminformers.InformerKeyEgress) {
+		egressPolicy := egressIface.(*policyv1alpha1.Egress)
+
+		if !c.IsMonitoredNamespace(egressPolicy.Namespace) {
+			continue
+		}
+		policies = append(policies, egressPolicy)
+	}
+
+	return policies
+}
+
+// GetIngressBackendPolicy returns the IngressBackend policy for the given backend MeshService
+func (c *Client) GetIngressBackendPolicy(namespace, svcName string, targetPort int) *policyv1alpha1.IngressBackend {
+	for _, ingressBackendIface := range c.informers.List(osminformers.InformerKeyIngressBackend) {
+		ingressBackend := ingressBackendIface.(*policyv1alpha1.IngressBackend)
+
+		if ingressBackend.Namespace != namespace {
+			continue
+		}
+
+		// Return the first IngressBackend corresponding to the given MeshService.
+		// Multiple IngressBackend policies for the same backend will be prevented
+		// using a validating webhook.
+		for _, backend := range ingressBackend.Spec.Backends {
+			// we need to check ports to allow ingress to multiple ports on the same svc
+			if backend.Name == svcName && backend.Port.Number == int(targetPort) {
+				return ingressBackend
+			}
+		}
+	}
+
+	return nil
+}
+
+func (c *Client) ListIngressBackendPolicies() []*policyv1alpha1.IngressBackend {
+	var backends []*policyv1alpha1.IngressBackend
+
+	for _, ingressBackendIface := range c.informers.List(osminformers.InformerKeyIngressBackend) {
+		backend := ingressBackendIface.(*policyv1alpha1.IngressBackend)
+		if !c.IsMonitoredNamespace(backend.Namespace) {
+			continue
+		}
+
+		backends = append(backends, backend)
+	}
+
+	return backends
+}
+
+// ListRetryPolicies returns the retry policies for the given source identity based on service accounts.
+func (c *Client) ListRetryPolicies() []*policyv1alpha1.Retry {
+	var retries []*policyv1alpha1.Retry
+
+	for _, retryInterface := range c.informers.List(osminformers.InformerKeyRetry) {
+		policy := retryInterface.(*policyv1alpha1.Retry)
+		if !c.IsMonitoredNamespace(policy.Namespace) {
+			continue
+		}
+
+		retries = append(retries, policy)
+	}
+
+	return retries
+}
+
+// ListUpstreamTrafficSettings returns the all UpstreamTrafficSetting resources
+func (c *Client) ListUpstreamTrafficSettings() []*policyv1alpha1.UpstreamTrafficSetting {
+	var settings []*policyv1alpha1.UpstreamTrafficSetting
+
+	// Filter by MeshService
+	for _, resource := range c.informers.List(osminformers.InformerKeyUpstreamTrafficSetting) {
+		setting := resource.(*policyv1alpha1.UpstreamTrafficSetting)
+
+		if !c.IsMonitoredNamespace(setting.Namespace) {
+			continue
+		}
+
+		settings = append(settings, setting)
+	}
+
+	return settings
+}
+
+// GetUpstreamTrafficSettingByNamespace returns the UpstreamTrafficSetting resources with matching namespaced name
+func (c *Client) GetUpstreamTrafficSettingByKey(key string) *policyv1alpha1.UpstreamTrafficSetting {
```

Comment:
super nit: instead of key, let's pass in name, namespace and construct the key in here

Reply:
I just noticed that the caller of this method uses a `types.NamespacedName` obj, which is k8s native object. So it should be fine to pass the object to this method